massa-wallet/src/lib.rs

```diff
@@ -142,13 +140,18 @@ impl Wallet {
         let cipher = Aes256GcmSiv::new(Key::from_slice(
             Hash::compute_from(self.password.as_bytes()).to_bytes(),
         ));
-        let nonce = Nonce::from_slice(NONCE_SLICE);
+        let mut bytes = [0u8; 12];
+        thread_rng().fill_bytes(&mut bytes);
+        let nonce = Nonce::from_slice(&bytes);
         let keys =
             serde_json::to_string(&self.keys.iter().map(|(_, (_, pk))| *pk).collect::<Vec<_>>())?;
         let text = cipher
             .encrypt(nonce, keys.as_ref())
             .map_err(|e| WalletError::EncryptionError(e.to_string()))?;
-        std::fs::write(&self.wallet_path, text)?;
+        let mut content = bytes.to_vec();
+        content.extend(b":");
+        content.extend(text);
```

Comment:
note: technically the variable name is wrong (it's not text but encrypted bytes)